test/onnx/test_diagnostics.py

```diff
@@ -0,0 +1,239 @@
+# Owner(s): ["module: onnx"]
+
+import contextlib
+import io
+from typing import Set, Tuple
+
+import torch
+from torch.onnx import errors
+from torch.onnx._internal import diagnostics
+from torch.onnx._internal.diagnostics import infra
+from torch.testing._internal import common_utils
+
+
+def _assert_has_diagnostics(
+    engine: infra.DiagnosticEngine,
+    rule_level_pairs: Set[Tuple[infra.Rule, infra.Level]],
```

Comment:
```suggestion
    rule_level_pairs: AbstractSet[Tuple[infra.Rule, infra.Level]],
```
https://gist.github.com/justinchuby/4021cebe9e093f636759a88de325c85f